hal/src/electron/modem/electronserialpipe_hal.cpp

```diff
@@ -135,6 +112,52 @@ void ElectronSerialPipe::begin(unsigned int baud)
     USART_ITConfig(USART3, USART_IT_RXNE, ENABLE);
 }
 
+void ElectronSerialPipe::end()
+{
+    if (baud_ == 0) {
+        return;
+    }
+
+    // wait for transmission of outgoing data
+    while (_pipeTx.readable())
+    {
+        char c = _pipeTx.getc();
+        USART_SendData(USART3, c);
```

Comment:
This will overflow the TX buffer in USART peripheral. I see two possible options here:
```c++
txStart();
while (_pipeTx.readable()) {
  os_thread_yield();
}
```

Or if the idea is not to use interrupts here this loop should instead be transformed into:
```c++
USART_ITConfig(USART3, USART_IT_TXE, DISABLE);
while (_pipeTx.readable()) {
  txCopy();
}
```